Tests/MongoSwiftTests/CollectionTests.swift

```diff
@@ -298,4 +298,26 @@ final class CollectionTests: XCTestCase {
         }
     }
 
+    func testEncodeCursorType() throws {
+        let encoder = BsonEncoder()
+
+        let nonTailable = FindOptions(cursorType: .nonTailable)
+        expect(try encoder.encode(nonTailable)).to(equal(["awaitData": false, "tailable": false]))
+
+        let tailable = FindOptions(cursorType: .tailable)
+        expect(try encoder.encode(tailable)).to(equal(["awaitData": false, "tailable": true ]))
+
+        let tailableAwait = FindOptions(cursorType: .tailableAwait)
+        expect(try encoder.encode(tailableAwait)).to(equal(["awaitData": true, "tailable": true ]))
+    }
+
+    func testEncodeHint() throws {
+        let encoder = BsonEncoder()
+
+        let stringHint = AggregateOptions(hint: .indexName("hi"))
+        expect(try encoder.encode(stringHint)).to(equal(["hint": "hi"]))
+
+        let docHint = AggregateOptions(hint: .indexSpec(["hi": 1]))
+        expect(try encoder.encode(docHint)).to(equal(["hint": ["hi": 1] as Document]))
```

Comment:
This casts `["hi": 1]` as a Document, but don't you also want `["hint": ["hi": 1] as Document]` cast to a Document? Is that omitted because you're not testing that AggregateOptions itself is encoded to a Document here (and/or just doing a less strict equality check)?

Reply:
so the compiler is able to figure out that the whole thing (`["hint": ["hi": 1] as Document]`) is a `Document` because I'm comparing it to the result of a function that always returns a `Document` (`encode`). 

the reason I have to say `as Document` for a nested doc is that the compiler *doesn't* know that the only `BsonValue` that is `ExpressibleByDictionaryLiteral` is a `Document`. so it considers just saying `["hint": ["hi": 1]]` to be ambiguous